Tests/OpenFeatureTests/ProviderEventsTests.swift

```diff
@@ -19,4 +19,20 @@ final class ProviderEventsTests: XCTestCase {
         wait(for: [readyExpectation], timeout: 5)
         XCTAssertNotNil(eventState)
     }
+
+    func testErrorEventSent() {
+        provider.initWithError()
+        let errorExpectation = XCTestExpectation(description: "Error")
+        let eventState = provider
+            .observe()
+            .filter { event in
+                event == ProviderEvent.error
+            }
+            .sink { _ in
+                errorExpectation.fulfill()
+            }
+        OpenFeatureAPI.shared.setProvider(provider: provider)
```

Comment:
I believe what we really want to test in this PR is that when we call `SetProviderAndAwait` we will get to the next line (basically returning from the function).